### PR TITLE
fix: enc/dec rlp for `Requests`

### PR DESCRIPTION
**Pull request by onbjerg**
this is cursed

we need to encode `Requests` as `rlp([req_bytes_1, req_bytes_2, ..])`, but previously we were encoding it as `rlp([rlp(req_bytes_1), ..])`

### CODE REVIEW

File: crates/primitives/src/request.rs [hunk 1]
```diff
@@ -23,3 +25,25 @@ impl IntoIterator for Requests {
...
+}
+
+impl Decodable for Requests {
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        Ok(Self(
+            <Vec<Bytes> as Decodable>::decode(buf)?
+                .into_iter()
+                .map(|bytes| Request::decode_7685(&mut bytes.as_ref()))
+                .collect::<Result<Vec<_>, alloy_eips::eip7685::Eip7685Error>>()?,
+        ))
```
Comment (onbjerg) [inline crates/primitives/src/request.rs:47]: @DaniPopes how to make this cool

Comment (DaniPopes) [reply crates/primitives/src/request.rs:47]: looks fine, maybe map(Self) at the end

File: crates/primitives/src/request.rs [hunk 2]
```diff
@@ -23,3 +25,25 @@ impl IntoIterator for Requests {
...
 }
+
+impl Encodable for Requests {
+    fn encode(&self, out: &mut dyn bytes::BufMut) {
+        self.0
+            .iter()
+            .map(Encodable7685::encoded_7685)
+            .map(Bytes::from)
+            .collect::<Vec<_>>()
+            .encode(out)
```
Comment (mattsse) [inline crates/primitives/src/request.rs:36]: I don't think we need the alloc here and can just mimic the vec encode impl

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/primitives/src/request.rs (hunk 2 above)
Comment (onbjerg) [reply crates/primitives/src/request.rs]: need an alloc regardless, made it more cursed in [7500152](https://github.com/paradigmxyz/reth/pull/8271/commits/7500152ce2a6448191da74f3b53c028cb4d6d237), feel free to give it a shot

**Review by gakonst — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
pending @DaniPopes

**Review by DaniPopes — APPROVED** [review]
lg mt

**Pull request merged**